msal/src/main/java/com/microsoft/identity/client/IPublicClientApplication.java

```diff
@@ -162,4 +176,29 @@ void acquireToken(@NonNull final Activity activity,
         void onError(final MsalException exception);
     }
 
+    interface DeviceCodeFlowCallback{
+        /**
+         * Called to display verification uri, user code, and instruction message during device code flow.
+         *
+         * @param vUri verification uri
+         * @param userCode user code
+         * @param message instruction message
+         */
+        void getUserCode(@NonNull String vUri, @NonNull String userCode, @NonNull String message);
+
+        /**
+         * Called once succeed and pass the result object.
+         *
+         * @param authResult the authentication result
+         */
+        void getToken(AuthenticationResult authResult);
```

Comment:
Doesn't seem like it returns anything so I think a name that start with `get` is not suitable. How about `onTokenReceived`?